### PR TITLE
Add a button to allow resetting the ComboboxControl value

### PR DESCRIPTION
**Pull request by youknowriad**
When using the ComboboxControl in selects where "null" values are allowed like #25267 we need to be able to reset its value.
This PR does so by adding a reset button.

**Testing instructions**

 - npm run storybook:dev
 - Check the ComboboxControl story and see that you can choose countries and reset the value as well.

### CODE REVIEW

File: packages/components/src/combobox-control/index.js [hunk 1]
```diff
@@ -158,21 +165,38 @@ function ComboboxControl( {
...
+							isExpanded={ isExpanded }
+							selectedSuggestionIndex={ matchingSuggestions.indexOf(
+								selectedSuggestion
+							) }
+							onChange={ onInputChange }
+						/>
+					</FlexItem>
+					{ allowReset && (
+						<FlexItem>
+							<Flex>
```
Comment (youknowriad) [inline packages/components/src/combobox-control/index.js:188]: This extra Flex was only needed to center the Button properly, I was wondering if FlexItems should automatically be flexes as well 🤔 cc @ItsJonQ

File: packages/components/src/flex/styles/flex-styles.js [hunk 2]
```diff
@@ -60,17 +60,19 @@ const reversedStyles = ( { isReversed } ) => {
...
 	${ gapStyle };
 	${ reversedStyles };
 `;
 
-export const Item = styled.div`
+export const Item = styled.dev`
 	box-sizing: border-box;
 	min-width: 0;
 	max-width: 100%;
+	flex-grow: ${ ( props ) => props.grow };
```
Comment (youknowriad) [inline packages/components/src/flex/styles/flex-styles.js:75]: I hesitated between a "grow" prop or an "expand" one.

**Comment by ItsJonQ** [comment]
@youknowriad Haiii! So cool that you're using `Flex` in this component 😍 .

You can achieve what you need with `FlexBlock`. It's designed to take up the rest of the remaining space.

```jsx
<Flex>
	<FlexBlock>Adaptively wide thing</FlexBlock>
	<FlexItem>Tiny thing</FlexItem>
</Flex>
```

The only thing I needed to adjust (css wise) was to change the `display` of the button to `flex`.
The reason for this is because `inline-flex` (it's original `display`) causes it to have bottom `margin` styles.
This is the same for `display: inline` as well.

Switching it to a block based HTML element `display: flex | block` resolves this 👍 

Hope this helps!!

File: packages/components/src/flex/styles/flex-styles.js [hunk 3]
```diff
@@ -60,11 +60,12 @@ const reversedStyles = ( { isReversed } ) => {
 export const Flex = styled.div`
 	box-sizing: border-box;
 	display: flex;
+	width: 100%;
 
-	${ alignStyle };
-	${ justifyStyle };
-	${ gapStyle };
-	${ reversedStyles };
+	${ alignStyle }
```
Comment (ItsJonQ) [inline packages/components/src/flex/styles/flex-styles.js:65]: This is from auto formatting 🤷‍♀️

File: packages/components/src/flex/styles/flex-styles.js (hunk 2 above)
Comment (mcsf) [reply packages/components/src/flex/styles/flex-styles.js]: I prefer "grow" since it's already CSS/flex terminology.

**Review by mcsf — APPROVED** [review]

**Comment by ItsJonQ** [comment]
Thanks @mcsf ! Looks like there's a snapshot test conflict. I'm going to resolve that and merge :)

**Pull request merged**